crate/operator/utils/kubeapi.py

```diff
@@ -191,18 +192,13 @@ async def get_host(core: CoreV1Api, namespace: str, name: str) -> str:
         # During testing we need to connect to the cluster via its public IP
         # address, because the operator isn't running inside the Kubernetes
         # cluster.
-        logger = logging.getLogger(__name__)
-        from crate.operator.create import recreate_services
-        from crate.operator.operations import (
-            get_cratedb_resource,
-            is_lb_service_present,
+        services = await core.list_namespaced_service(namespace)
+        service = next(
+            (svc for svc in services.items if svc.metadata.name == f"crate-{name}"),
+            None,
         )
-
-        if not await is_lb_service_present(core, namespace, name):
-            cratedb = await get_cratedb_resource(namespace, name)
-            await recreate_services(
-                namespace, name, cratedb["spec"], cratedb["metadata"], logger
-            )
+        if not service:
+            raise kopf.TemporaryError("Waiting for service to be created...", delay=5)
```

Comment:
Maybe better place would be to throw this exception in `get_service_public_hostname`, and  get rid of the for loop there?

Reply:
Yes I thought so too, unfortunately `get_service_public_hostname` is also used outside `kopf` [here](https://github.com/crate/crate-operator/blob/master/tests/utils.py#L181) which makes all tests blow up when I raise a `kopf.TemporaryError` in there :thinking:

Reply:
You could make a for loop there instead? :stuck_out_tongue: